packages/dataviews/src/view-table.js

```diff
@@ -283,6 +295,19 @@ function ViewTable( {
 	isLoading = false,
 	deferredRendering,
 } ) {
+	const headerMenuRefs = useRef( {} );
+	const onHide = ( field ) => {
+		const hidden = headerMenuRefs.current[ field.id ];
+		const fallback = headerMenuRefs.current[ hidden.fallback ];
+		// These stacked microtask callbacks are required to
+		// make sure the node is available to be focused
+		queueMicrotask( () => {
+			queueMicrotask( () => {
+				if ( fallback?.node ) fallback.node.focus();
+			} );
+		} );
```

Comment:
I am uneasy with this use of `queueMicrotask`, as it relies on an incidental fact that we currently need to wait two re-renders, which I'm not sure is not guaranteed to be true forever.

Have we considered other ways to achieve this, ideally in a more explicit manner? Here is one way I can think of, but there may be better ways:

```js
	const [ headerFieldToFocus, setHeaderFieldToFocus ] = useState( null );
	const onHide = ( field ) => {
		const hidden = headerMenuRefs.current[ field.id ];
		const fallback = headerMenuRefs.current[ hidden.fallback ];
		if ( fallback?.node ) {
			setHeaderFieldToFocus( fallback.node );
		}
	};

	headerFieldToFocus?.focus();
	if (
		headerFieldToFocus?.ownerDocument.activeElement === headerFieldToFocus
	) {
		setHeaderFieldToFocus( null );
	}
```

Reply:
Yeah, it's less than ideal, and you're right, we shouldn't be relying on incidental behaviour. Will revisit.